lib/workflow/messengers/messenger-AMQP.js

```diff
@@ -84,79 +83,62 @@ function amqpMessengerFactory(
      * @returns {Promise}
      */
     exports.publishProgressEvent = function(graphId, data) {
-
-        assert.uuid(graphId, "Progress data should include graphId");
-        assert.object(data.progress, "Progress data should include progress info");
-
-        return Promise.resolve()
-        .then(function(){
-            //Update graphName if it doesn't exist in data
-            return waterline.graphobjects.findOne({instanceId: graphId})
-            .then(function(graphObject){
-                if(!_.has(data, 'graphName') &&
-                   _.has(graphObject, 'definition.friendlyName')) {
-                    data.graphName = graphObject.definition.friendlyName;
-                }
-                return graphObject;
-            });
+        return Promise.try(function(){
+            assert.uuid(graphId, "Progress data should include graphId");
+            assert.object(data.progress, "Progress data should include progress info");
+            return waterline.graphobjects.findOne({instanceId: graphId});
         })
```

Comment:
Since you didn't deal with assert exceptions in EventsProtocol.prototype.publishProgressEvent above, I think we can leave the exceptions to be done with caller's promise.